liota/lib/transports/mqtt.py

```diff
@@ -215,51 +230,79 @@ def connect_soc(self):
-                self._paho_client.tls_set(self.identity.root_ca_cert,
-                                          cert_reqs=getattr(ssl, self.tls_conf.cert_required),
-                                          tls_version=getattr(ssl, self.tls_conf.tls_version),
-                                          ciphers=self.tls_conf.cipher)
+                context.load_cert_chain(self.identity.cert_file, self.identity.key_file)
             elif not client_cert_available and client_key_available:
-                log.error("Error : Client key found, but client certificate not found")
                 raise ValueError("Error : Client key found, but client certificate not found")
-            else:
-                log.error("Error : Client key found, but client certificate not found")
+            elif client_cert_available and not client_key_available:
                 raise ValueError("Error : Client certificate found, but client key not found")
-            log.info("TLS support is set up.")
+            else:
+                log.info("Client Certificate and Client Key are not provided")
+
+            if getattr(ssl, self.tls_conf.cert_required) == ssl.CERT_NONE and hasattr(context, 'check_hostname'):
+                context.check_hostname = False
+
+            context.verify_mode = ssl.CERT_REQUIRED if self.tls_conf.cert_required is None else getattr(ssl,
+                                                                                                        self.tls_conf.cert_required)
+
+            if self.identity.root_ca_cert is not None:
+                context.load_verify_locations(self.identity.root_ca_cert)
+            else:
+                context.load_default_certs()
+
+            if self.tls_conf.cipher is not None:
+                context.set_ciphers(self.tls_conf.ciphers)
+
+            # Setting the verify_flags to VERIFY_CRL_CHECK_CHAIN in this mode
+            # certificate revocation lists (CRLs) of all certificates in the
+            # peer cert chain are checked if the path of CRLs in PEM or DER format
+            # is specified
+
+            if self.tls_conf.crl_path is not None:
+                if os.path.exists(self.tls_conf.crl_path):
+                    context.verify_flags = ssl.VERIFY_CRL_CHECK_CHAIN
+                    context.load_verify_locations(cafile=self.tls_conf.crl_path)
+                else:
+                    raise ValueError("Error : Wrong Client CRL path")
+
+            # Setting the tls context
+            self._paho_client.tls_set_context(context)
+
+            if getattr(ssl, self.tls_conf.cert_required) != ssl.CERT_NONE:
+                # Default to secure, sets context.check_hostname attribute
+                # if available
+                self._paho_client.tls_insecure_set(False)
+            else:
+                # But with ssl.CERT_NONE, we can not check_hostname
+                self._paho_client.tls_insecure_set(True)
```

Comment:
How does CERT_NONE get set?

Reply:
It is passed from the user code by setting a parameter in sampleProp.conf

https://github.com/vmware/liota/blob/master/packages/sampleProp.conf#L34